source/extensions/filters/http/jwt_authn/verifier.cc

```diff
@@ -307,11 +307,10 @@ class AnyVerifierImpl : public BaseGroupVerifierImpl {
       // it should be used as the final status.
       Status final_status = Status::JwtMissed;
       for (const auto& it : verifiers_) {
-        // If a Jwt is extracted from a location not specified by the required provider,
-        // the authenticator returns JwtUnknownIssuer. It should be treated the same as
-        // JwtMissed.
+        // Prefer to return non-JwtMissed and non-JwtUnknownIssuer error.
         Status child_status = context.getCompletionState(it.get()).status_;
-        if (child_status != Status::JwtMissed && child_status != Status::JwtUnknownIssuer) {
+        if ((child_status != Status::JwtMissed && child_status != Status::JwtUnknownIssuer) ||
+            final_status == Status::JwtMissed) {
```

Comment:
this means no matter what the first child_status is, we will always assign it to final_status? Is this expected? (I know this fixed the issuer but not sure if it may have other side effects).

Reply:
no, child_status could be
1) Missed
2) UnknownIssuer
3) other errors

We prefer 3),  try to get last 3),  
otherwise, we keep 2). 

That is the logic here.

Reply:
I see, could you update the comment in line 310 for this, I think between JwtMissed and JwtUnknownIssuer, we prefer the JwtUnknownIssuer, otherwise LGTM.

Reply:
Updated the comment. Thanks